### PR TITLE
Add jetson build on CI

### PR DESCRIPTION
**Pull request by lanluo-nvidia**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: packaging/pre_build_script.sh [hunk 1]
```diff
@@ -5,60 +5,104 @@ set -x
...
-         -e "s/tensorrt-cu12-bindings/tensorrt-${CU_VERSION::4}-bindings/g" \
-         -e "s/tensorrt-cu12-libs/tensorrt-${CU_VERSION::4}-libs/g" \
-         pyproject.toml
-fi
+export TORCH_BUILD_NUMBER=$(python -c "import torch, urllib.parse as ul; print(ul.quote_plus(torch.__version__))")
+export TORCH_INSTALL_PATH=$(python -c "import torch, os; print(os.path.dirname(torch.__file__))")
 
-if [[ ${TENSORRT_VERSION} != "" ]]; then
-  cat toolchains/ci_workspaces/MODULE_tensorrt.bazel.tmpl | envsubst > MODULE.bazel
+export CUDA_NAME=cuda
```
Comment (narendasan) [inline packaging/pre_build_script.sh:60]: I think a lot of this stuff does not need to be templatized, basically just versions and dep urls. Doing dep updates by editing this script seems complicated and potentially error prone

File: docsrc/getting_started/jetpack.rst [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+.. _Torch_TensorRT_in_JetPack_6.2
 
 Overview
 ##################
 
-JetPack 6.1
+JetPack 6.2
 ---------------------
-Nvida JetPack 6.1 is the latest production release ofJetPack 6.
+Nvida JetPack 6.2 is the latest production release of JetPack 6.
```
Comment (narendasan) [inline docsrc/getting_started/jetpack.rst:8]: NVIDIA

File: docsrc/getting_started/jetpack.rst [hunk 3]
```diff
@@ -63,17 +63,30 @@ Ensure libcusparseLt.so exists at /usr/local/cuda/lib64/:
     sudo cp -a libcusparse_lt-linux-sbsa-0.5.2.1-archive/lib/* /usr/local/cuda/lib64/
 
 
-Build torch_tensorrt
-~~~~~~~~~~~~~~
+Build torch_tensorrt wheel file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (narendasan) [inline docsrc/getting_started/jetpack.rst:68]: This section should focus on source builds on device, I dont think we want to recommend the docker container as default

File: packaging/pre_build_script.sh [hunk 4]
```diff
@@ -5,60 +5,102 @@ set -x
...
 
+export TORCH_BUILD_NUMBER=$(python -c "import torch, urllib.parse as ul; print(ul.quote_plus(torch.__version__))")
+export TORCH_INSTALL_PATH=$(python -c "import torch, os; print(os.path.dirname(torch.__file__))")
+
+echo "TORCH_BUILD_NUMBER=$TORCH_BUILD_NUMBER"
+echo "TORCH_INSTALL_PATH=$TORCH_INSTALL_PATH"
+cat toolchains/ci_workspaces/MODULE.bazel.tmpl | envsubst > MODULE.bazel
+
+if [[ ${TENSORRT_VERSION} != "" ]]; then
+    sed -i -e "s/strip_prefix = \"TensorRT-.*\"/strip_prefix = \"${TENSORRT_STRIP_PREFIX}\"/g" MODULE.bazel
```
Comment (narendasan) [inline packaging/pre_build_script.sh:101]: Cant we just use envsubst for this?

Comment (narendasan) [reply packaging/pre_build_script.sh:101]: Is this for the TRT future tests?

Comment (lanluo-nvidia) [reply packaging/pre_build_script.sh:101]: yes.

File: docsrc/getting_started/jetpack.rst [hunk 5]
```diff
@@ -1,119 +1,133 @@
...
 .. code-block:: sh
 
-    # install pip
-    wget https://bootstrap.pypa.io/get-pip.py
-    python get-pip.py
+   # Configure build environment
+   export BUILD_VERSION="2.8.0.dev20250511"
+   export CUDA_HOME=/usr/local/cuda-12.6
+   export TORCH_INSTALL_PATH=$(python -c "import torch, os; print(os.path.dirname(torch.__file__))")
+   cat toolchains/ci_workspaces/MODULE.bazel.tmpl | envsubst > MODULE.bazel
```
Comment (narendasan) [inline docsrc/getting_started/jetpack.rst:108]: Why do we need this if we already have L4T sources in the main MODULE.bazel?

Comment (lanluo-nvidia) [reply docsrc/getting_started/jetpack.rst]: yes, User can use the main MODULE.bazel:
in this case we still need to change the path:
```
new_local_repository(
    name = "libtorch",
# please modify this path according to your local python wheel path
    path = "/usr/local/lib/python3.11/dist-packages/torch",
    build_file = "third_party/libtorch/BUILD"
)
```
in jetpack,change path from:
/usr/local/lib/python3.11/dist-packages/torch
to local python wheel path
according to 
python -c "import torch, os; print(os.path.dirname(torch.__file__))"

File: docsrc/getting_started/jetpack.rst [hunk 6]
```diff
@@ -1,119 +1,159 @@
...
+      python -m pip install pyyaml
 
-Ensure libcusparseLt.so exists at /usr/local/cuda/lib64/:
+3. **Install PyTorch**:
 
-.. code-block:: sh
+   .. code-block:: sh
+      
+      # Can only install the torch and torchvision wheel from the JPL repo which is built specifically for JetPack 6.2
+      TORCH_URL=https://pypi.jetson-ai-lab.dev/jp6/cu126/+f/6ef/f643c0a7acda9/torch-2.7.0-cp310-cp310-linux_aarch64.whl#sha256=6eff643c0a7acda92734cc798338f733ff35c7df1a4434576f5ff7c66fc97319
```
Comment (narendasan) [inline docsrc/getting_started/jetpack.rst]: These will go stale, lets use `pip install torch torchvision --index-url=https://pypi.jetson-ai-lab.dev/jp6/cu126/`

File: packaging/pre_build_script.sh [hunk 7]
```diff
@@ -5,32 +5,55 @@ set -x
...
-TORCH_TORCHVISION=$(grep "^torch" py/requirements.txt)
-INDEX_URL=https://download.pytorch.org/whl/${CHANNEL}/${CU_VERSION}
-
-# Install all the dependencies required for Torch-TensorRT
 pip uninstall -y torch torchvision
-pip install --force-reinstall --pre ${TORCH_TORCHVISION} --index-url ${INDEX_URL}
+
+if [[ ${IS_JETPACK} == true ]]; then
+    # install torch 2.7 torchvision 0.22.0 for jp6.2
+    TORCH_URL=https://pypi.jetson-ai-lab.dev/jp6/cu126/+f/6ef/f643c0a7acda9/torch-2.7.0-cp310-cp310-linux_aarch64.whl#sha256=6eff643c0a7acda92734cc798338f733ff35c7df1a4434576f5ff7c66fc97319
```
Comment (narendasan) [inline packaging/pre_build_script.sh]: Same here can we just use the index url?

File: packaging/pre_build_script.sh [hunk 8]
```diff
@@ -5,32 +5,52 @@ set -x
...
-TORCH_TORCHVISION=$(grep "^torch" py/requirements.txt)
-INDEX_URL=https://download.pytorch.org/whl/${CHANNEL}/${CU_VERSION}
-
-# Install all the dependencies required for Torch-TensorRT
 pip uninstall -y torch torchvision
-pip install --force-reinstall --pre ${TORCH_TORCHVISION} --index-url ${INDEX_URL}
+
+if [[ ${IS_JETPACK} == true ]]; then
+    # install torch 2.7 torchvision 0.22.0 for jp6.2
+    pip install torch==2.7.0 torchvision==0.22.0  --index-url=https://pypi.jetson-ai-lab.dev/jp6/cu126/
```
Comment (narendasan) [inline packaging/pre_build_script.sh:46]: what happens if the version is unconstrained? shouldnt it just take latest?

Comment (narendasan) [reply packaging/pre_build_script.sh:46]: That way we dont need to update it

File: .github/workflows/build-test-linux-x86_64.yml [hunk 9]
```diff
@@ -1,7 +1,7 @@
-name: Build and test Linux wheels
+name: Build and test Linux x86_64 wheels
 
 on:
-  pull_request:
+  #pull_request:
```
Comment (narendasan) [inline .github/workflows/build-test-linux-x86_64.yml]: Does this need to be merged?

File: .github/workflows/build-test-windows.yml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 name: Build and test Windows wheels
 
 on:
-  pull_request:
+  #pull_request:
```
Comment (narendasan) [inline .github/workflows/build-test-windows.yml]: Same here?

File: packaging/pre_build_script.sh (hunk 8 above)
Comment (lanluo-nvidia) [reply packaging/pre_build_script.sh:46]: the version can be unconstrained, pick the latest only if the pyproject.toml has the same version.

File: docsrc/getting_started/jetpack.rst [hunk 11]
```diff
@@ -1,119 +1,122 @@
...
 
-Build and Install torch_tensorrt wheel file
+Installation
+============
 
+.. code-block:: sh
 
-Since torch_tensorrt version has dependencies on torch version. torch version supported by JetPack6.1 is from DLFW 24.08/24.09(torch 2.5.0).
+   cd dist
+   python -m pip install torch-tensorrt-2.8.0.dev0+4da152843-cp310-none-linux_aarch64.whl
```
Comment (narendasan) [inline docsrc/getting_started/jetpack.rst]: Shouldnt this be the linux tegra version now?

**Review by narendasan — APPROVED** [review]
LGTM after minor docs update
